front/app/containers/Admin/settings/customize/Header/index.tsx

```diff
@@ -163,37 +216,92 @@ const Header = ({
     });
   };
 
+  const handleHeaderBgPreviewOnChange = (option: IOption) => {
+    setPreviewDevice(option.value);
+  };
+
+  const latestAppConfigCoreSettings = latestAppConfigSettings?.core;
+
   return (
     <Section key={'header'}>
       <SectionTitle>
         <FormattedMessage {...messages.header} />
       </SectionTitle>
-      <SectionField key={'header_bg'}>
-        <Label htmlFor="landingpage-header-dropzone">
-          <FormattedMessage {...messages.header_bg} />
-          <IconTooltip
-            content={<FormattedMessage {...messages.header_bgTooltip} />}
-          />
-        </Label>
-        <ImagesDropzone
-          id="landingpage-header-dropzone"
-          acceptedFileTypes="image/jpg, image/jpeg, image/png, image/gif"
-          images={header_bg}
-          imagePreviewRatio={480 / 1440}
-          maxImagePreviewWidth="500px"
+      <SectionDescription>
+        <FormattedMessage {...messages.headerDescription} />
+      </SectionDescription>
+      <Outlet
+        id="app.containers.Admin.settings.customize.headerSectionStart"
+        latestAppConfigSettings={latestAppConfigSettings}
+        handleOnChange={handleSettingOnChange}
+      />
+      <SubSectionTitle>
+        <FormattedMessage {...messages.header_bg} />
+      </SubSectionTitle>
+      <SectionField>
+        {header_bg && (
+          <>
+            <Label>
+              <FormattedMessage {...messages.bgHeaderPreviewSelectLabel} />
+            </Label>
+            <BgHeaderPreviewSelect
+              options={[
+                { value: 'desktop', label: 'Desktop' },
+                { value: 'tablet', label: 'Tablet' },
+                { value: 'phone', label: 'Phone' },
+              ]}
+              onChange={handleHeaderBgPreviewOnChange}
+              value={previewDevice}
+            />
+          </>
+        )}
+
+        <HeaderImageDropzone
           onAdd={handleHeaderBgOnAdd}
           onRemove={handleHeaderBgOnRemove}
-          errorMessage={headerError}
+          latestAppConfigStyleSettings={latestAppConfigStyleSettings}
+          headerError={headerError}
+          header_bg={header_bg}
+          previewDevice={previewDevice}
+          layout={layout}
         />
       </SectionField>
 
-      <Outlet
-        id="app.containers.Admin.settings.customize.fields"
-        onChange={handleAppConfigurationStyleChange}
-        theme={theme}
-        latestAppConfigStyleSettings={latestAppConfigStyleSettings}
-      />
-
+      {/* We only allow the overlay for the full-width banner layout for the moment. */}
```

Comment:
If you upload a picture switch to full-width banner, you can change the overlay color and opacity. But if you then switch to e.g. two columns, the overlay remains in the preview. Is this a bug or a feature?